runelite-client/src/test/java/net/runelite/client/plugins/opponentinfo/OpponentInfoPluginTest.java

```diff
@@ -0,0 +1,199 @@
+
+        // the npc attacking the player that is not the current opponent attacks a different player
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(aggro, otherPlayer));
+
+        // verify that the current opponent is still the npc the player is attacking
+        assertSame(opponentInfoPlugin.getLastOpponent(), attacked);
+
+        // the npc the player is attacking, the current opponent, attacks a different player
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(attacked, otherPlayer));
+
+        // verify that the current opponent is still the npc the player is attacking
+        assertSame(opponentInfoPlugin.getLastOpponent(), attacked);
+    }
+
+    /*
+     * Verify that the current opponent is the expected npc while the player is not attacking anything
+     */
+    @Test
+    public void testIdleInMulti() {
+        Player thePlayer = mock(Player.class), otherPlayer = mock(Player.class);
+        NPC aggroFirst = mock(NPC.class), aggroRecent = mock(NPC.class);
+        when(client.getLocalPlayer()).thenReturn(thePlayer);
+
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(thePlayer, null));
+
+        // verify that there is currently no opponent
+        assertNull(opponentInfoPlugin.getLastOpponent());
+
+        // some npc attacks the player
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(aggroFirst, thePlayer));
+
+        // verify that the current opponent is the aggressive npc
+        assertSame(opponentInfoPlugin.getLastOpponent(), aggroFirst);
+
+        // the player is attacked by a different npc
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(aggroRecent, thePlayer));
+
+        // verify that the current opponent is the most recent aggressor
+        assertSame(opponentInfoPlugin.getLastOpponent(), aggroRecent);
+
+        // an npc that is not the current opponent targets another player
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(aggroFirst, otherPlayer));
+
+        // verify that the current opponent is still the most recent aggressor
+        assertSame(opponentInfoPlugin.getLastOpponent(), aggroRecent);
+
+        // the current opponent switches targets to another player
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(aggroRecent, otherPlayer));
+
+        // verify that there is no longer an opponent
+        // (if lastTime is not null, then the lastOpponent will become null after OpponentInfoPlugin.WAIT time)
+        assertTrue(opponentInfoPlugin.getLastTime() != null || opponentInfoPlugin.getLastOpponent() == null);
+    }
+
+    /*
+     * Verify that the current opponent is the expected npc in singles
+     */
+    @Test
+    public void testSingles() {
+        Player thePlayer = mock(Player.class), otherPlayer = mock(Player.class);
+        NPC playerNPC = mock(NPC.class);
```

Comment:
This is a very confusing var name. Couldn't we just call this `npc`?

Reply:
You're right, at one point I was using two npcs and I just never changed the name when I removed one of them.